### PR TITLE
[time] sunpy.time to astropy time

### PR DESCRIPTION
**Pull request by vn-ki**
This PR should fix `sunpy.time`(`TimeRange`, `julian` and `utime`).

### CODE REVIEW

File: sunpy/time/timerange.py [hunk 1]
```diff
@@ -201,7 +195,8 @@ def __eq__(self, other):
         result : `bool`
         """
         if isinstance(other, TimeRange):
-            return (self.start == other.start) and (self.end == other.end)
+            return _is_time_equal(
```
Comment (Cadair) [inline sunpy/time/timerange.py]: This is a work around right? if so should have a comment

Comment (vn-ki) [reply sunpy/time/timerange.py]: Yeah. I'll add it.

File: sunpy/time/tests/test_astropy_time.py [hunk 2]
```diff
@@ -104,3 +107,27 @@ def test_strptime_leapsecond():
...
+    t2 = Time('1995-12-31T23:59:60', format='isot') + TimeDelta(0*u.day)
+
+    assert _is_time_equal(t1, t2)
+
+
+def test_is_time_equal_not_equal():
+    t1 = Time('1995-12-31T23:59:59', format='isot')
+    t2 = Time('1995-12-31T23:59:60', format='isot')
+
+    assert not _is_time_equal(t1, t2)
```
Comment (Punyaslok) [inline sunpy/time/tests/test_astropy_time.py]: Maybe another test similar to the `+ TimeDelta(0*u.day)` test. Check to see if fail happens if difference is ~1 ns ?

File: sunpy/time/tests/test_julian.py [hunk 3]
```diff
@@ -21,75 +21,87 @@
...
 def test_julian_day():
     assert julian.julian_day('1900-01-01 12:00') == 2415021.0
     assert julian.julian_day(LANDING) == 2439159.5
     result = julian.julian_day('2000-03-01 15:30:26')
     assert_almost_equal(result, 2451605.1461111, decimal=3)
 
+
 def test_julian_day1():
     """should return julian day for date 1"""
     expected_day = 2415021.0
```
Comment (Punyaslok) [inline sunpy/time/tests/test_julian.py]: Either you use a newline after the doc string, or you don't. #ocd

File: sunpy/time/timerange.py (hunk 1 above)
Comment (vn-ki) [reply sunpy/time/timerange.py]: I added the comment to the doc string of the function.

File: sunpy/time/time.py [hunk 4]
```diff
@@ -273,24 +273,11 @@ def is_time(time_string, time_format=None):
     out : bool
         True if can be parsed by parse_time
 
-    Notes
-    -----
-        If time_string is an instance of float, then it is assumed to be in
-        unix time format.
```
Comment (nabobalis) [inline sunpy/time/time.py]: What is the case now?

File: sunpy/time/time.py [hunk 5]
```diff
@@ -273,24 +273,11 @@ def is_time(time_string, time_format=None):
...
-
     Examples
     --------
     >>> import sunpy.time
-    >>> sunpy.time.parse_time('2012/08/01')
-    datetime.datetime(2012, 8, 1, 0, 0)
-    >>> sunpy.time.parse_time('2005-08-04T00:01:02.000Z')
-    datetime.datetime(2005, 8, 4, 0, 1, 2)
```
Comment (nabobalis) [inline sunpy/time/time.py]: Do we want these examples removed?

Comment (vn-ki) [reply sunpy/time/time.py]: Yup. And proper ones were added in their place.

File: sunpy/time/time.py (hunk 4 above)
Comment (vn-ki) [reply sunpy/time/time.py]: You have to specify the format yourself with the `format` keyword.

File: sunpy/time/astropy_time.py [hunk 6]
```diff
@@ -127,7 +127,7 @@ def value(self):
         for jd, out in iterator:
             out[...] = timedelta(days=jd.item())
 
-        return self.mask_if_needed(iterator.operands[-1])
+        return iterator.operands[-1]
```
Comment (vn-ki) [inline sunpy/time/astropy_time.py]: Note to self: astropy 3.1 has `mask_if_needed` while 3.0 don't.

**Pull request merged**